### PR TITLE
Apim 3247 add button to open api in portal

### PR DESCRIPTION
**Pull request by jourdiw**
## Issue

https://gravitee.atlassian.net/browse/APIM-3247

## Description

Add button to Documentation title that opens the Api in the Portal in a different window.

Page list:

Screenshot 2023-11-20 at 17 15 51

Edit page:

Screenshot 2023-11-20 at 17 15 33

When API is not published:

Screenshot 2023-11-21 at 16 06 03

## Additional context

---

📚&nbsp;&nbsp;View the storybook of this branch [here](https://612657caa8e859003a8a6430-yojmvhbeuv.chromatic.com)

### CODE REVIEW

File: gravitee-apim-console-webui/src/management/api/documentation-v4/components/api-documentation-v4-page-title/api-documentation-v4-page-title.component.ts [hunk 1]
```diff
@@ -13,11 +13,30 @@
...
 
 @Component({
   selector: 'api-documentation-page-title',
   template: require('./api-documentation-v4-page-title.component.html'),
   styles: [require('./api-documentation-v4-page-title.component.scss')],
 })
-export class ApiDocumentationV4PageTitleComponent {}
+export class ApiDocumentationV4PageTitleComponent implements OnInit {
+  @Input()
+  apiId: string;
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/components/api-documentation-v4-page-title/api-documentation-v4-page-title.component.ts]: I don't know if it's better to use `ajsStateParams` or an `input` 🤔 🤷🏼

File: gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.component.spec.ts [hunk 2]
```diff
@@ -376,6 +390,26 @@ describe('ApiDocumentationV4', () => {
...
+  describe('Header', () => {
+    it('should display Open in Portal button', async () => {
+      await init([], []);
+      const header = await harnessLoader.getHarness(ApiDocumentationV4PageTitleHarness);
+      expect(header).toBeDefined();
+      const openInPortalBtn = await header.getOpenInPortalBtn();
+      expect(openInPortalBtn).toBeDefined();
+      expect(await openInPortalBtn.isDisabled()).toEqual(false);
+    });
+    it('should not display Open in Portal button if Portal url not defined', async () => {
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/api-documentation-v4.component.spec.ts]: when there is an expect in a catch I'm used to add an `expect.assertions(x);` at the beginning of the test. It's just to be sure the expect will be done. 

I'm even not sure that the `.then((_) => fail('Button should not be found'))`is necessary here

File: gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-edit-page/api-documentation-v4-edit-page.component.spec.ts [hunk 3]
```diff
@@ -546,6 +560,32 @@ describe('ApiDocumentationV4EditPageComponent', () => {
...
+    it('should not display Open in Portal button if Portal url not defined', async () => {
+      await init(undefined, PAGE.id, null);
+      initPageServiceRequests({ pages: [PAGE], breadcrumb: [], parentId: undefined, mode: 'edit' }, PAGE);
+
+      const header = await harnessLoader.getHarness(ApiDocumentationV4PageTitleHarness);
+      expect(header).toBeDefined();
+      await header
+        .getOpenInPortalBtn()
+        .then((_) => fail('Button should not be found'))
+        .catch((err) => expect(err).toBeTruthy());
```
Comment (Okhelifi) [inline gravitee-apim-console-webui/src/management/api/documentation-v4/documentation-edit-page/api-documentation-v4-edit-page.component.spec.ts]: Same here for the then and catch

File: gravitee-apim-console-webui/src/management/api/documentation-v4/components/api-documentation-v4-page-title/api-documentation-v4-page-title.component.ts (hunk 1 above)
Comment (jourdiw) [reply gravitee-apim-console-webui/src/management/api/documentation-v4/components/api-documentation-v4-page-title/api-documentation-v4-page-title.component.ts]: I put it as Input for it to be the most dummy possible and to be a real child component. But 🤷‍♀️

**Review by Okhelifi — APPROVED** [review]

**Pull request merged**